North-Seattle-College/ad440-winter2021-thursday-repo
# Delete User on UsersPaginated

Team 1
Sprint 5 [ BUG ] Update from PR [270](https://github.com/North-Seattle-College/ad440-winter2021-thursday-repo/pull/270)
Issue [#287](https://github.com/North-Seattle-College/ad440-winter2021-thursday-repo/issues/287)


# This PR is for the Delete functionality for the `/users` Endpoint
- the route has been changed to UserPaginated. The previous PR with the endpoint was no longer accessible. Therefore this PR includes the Delete User option in UserPaginated.js. 
- Once merged you can test it [here ](https://nscstrdevusw2thucommon.z5.web.core.windows.net/users)


# Testing: 
1. Navigate to the correct directory:
`- cd ui/react-ui`
2. Install all packages, modules, and dependencies:
`- npm install `
3.Start the app in the development model:
`- npm start `

3. Navigate to Endpoint 
 - /users
 - 
![image](https://user-images.githubusercontent.com/32443542/111742213-f71da180-8844-11eb-968c-53ef885786c2.png)

4. You will see the Delete Colum  on the left  
![image](https://user-images.githubusercontent.com/32443542/111740700-6e9e0180-8842-11eb-9186-f568c8563f5c.png)
 - if you want to delete a user you can click on the row 
5. A Modal will pop up  that will ask you to cancel or confirm to delete the user 
![image](https://user-images.githubusercontent.com/32443542/111740889-c0df2280-8842-11eb-9bc2-ed4ff6fccbdc.png)
6. If you Confirm the User will be removed from the row and a pop-up with successful deletion will appear
![image](https://user-images.githubusercontent.com/32443542/111741046-0b609f00-8843-11eb-9a84-e787727d3eda.png)
 

# Changes Made: 

Refactored: `ConfirmDeleteUserModal.js` from `Users.js`
Refactored: `ToastMessage.js ` fron `Users.js`
Updated: `UsersPaginated.js `

# TIME SPEND:

DATE | ACTIVITY | TIME
-- | -- | --
 03/17/2021| Review UsersPaginated Page and started implementing Delete User Option for UsersPaginated.js instead of Users.js | 2 hours
03/18/2021| implemented and tested Delete Option for UsersPaginated.js  | 1.5 hours
03/18/2021 | Wrote this PR | 30 min
 | TOTAL HOURS SPENT| 4hr
